### PR TITLE
Implement user management (create, list, get) via API

### LINKED ISSUE #3023: Surface user management via API client
**Is your feature request related to a problem? Please describe.**
I regularly want to create (and manage) users, via API, because I don't have direct CLI access to the server which hosts timesketch.

**Describe the solution you'd like**
I would like to have a feature, where I can create, list and modify users, when I have admin privileges within timesketch.

**Describe alternatives you've considered**
There is a feature that exists in `tsctl`, however, that limits user management to the time sketch server, prohibiting administration via API.

### PR DESCRIPTION
**Pull request by lo-chr**
This PR fixes most of the features, mentioned here: https://github.com/google/timesketch/issues/3023

This PR is a first step into further capabilities of user management via API, although it lacks the ability to manage a particular user itself.

I also added some features to the testing framework, enabling the test of "admin only" tests, via creating admin users during test setup.

**Checks**
- [x] All tests succeed -> there are some warnings in the Sigma Tests, which I did not touch. My added tests did succeed.
- [x] Unit tests added.

**Closing issues**
closes #3023

### CODE REVIEW

**Comment by lo-chr** [comment]
Giving this PR a bit more thought, I identified some security and data privacy issues. I will set this PR into draft mode again, fix the issues and submit the  PR again. Sorry for the inconvenience.

**Comment by lo-chr** [comment]
I changed the UserResource, so that it does not check for user privileges anymore.

Might be useful to limit details like mail address to admin users In the future.

**Comment by berggren** [comment]
Hi @lo-chr thanks for contributing! I'll review this on Friday, Jan 26.

File: api_client/python/timesketch_api_client/client.py [hunk 1]
```diff
@@ -451,6 +451,62 @@ def create_sketch(self, name, description=None):
...
+        for user_dict in response.get("objects", []):
+            user_id = user_dict["id"]
+            user_obj = user.User(user_id=user_id, api=self)
+            yield user_obj
+
+    def get_user(self, user_id):
+        """Get a sketch.
+
+        Args:
+            sketch_id: Primary key ID of the sketch.
```
Comment (berggren) [inline api_client/python/timesketch_api_client/client.py:503]:
```suggestion
            user_id: Primary key ID of the user.
```

File: api_client/python/timesketch_api_client/client.py [hunk 2]
```diff
@@ -451,6 +451,62 @@ def create_sketch(self, name, description=None):
...
+
+        Returns:
+            Instance of a Sketch object.
```
Comment (berggren) [inline api_client/python/timesketch_api_client/client.py:506]:
```suggestion
            Instance of a User object.
```

**Review by berggren — CHANGES_REQUESTED** [review]

**Comment by lo-chr** [comment]
I'll fix this over the weekend! Thank you for the review.

File: timesketch/api/v1/resources/user.py [hunk 3]
```diff
@@ -44,8 +46,100 @@ def get(self):
...
+            return_users.append(
+                {
+                    "id": user.id,
+                    "username": user.username,
+                    "name": user.name,
+                    "email": user.email,
+                    "active": user.active,
+                    "admin": user.admin,
+                }
+            )
```
[the remark this answers is not in the document]
Comment (lo-chr) [reply timesketch/api/v1/resources/user.py]: Changed this with https://github.com/google/timesketch/pull/3024/commits/bc0b8b383da04d1efcef072478f7824b6af539a1

File: timesketch/api/v1/resources/user.py [hunk 4]
```diff
@@ -46,6 +47,84 @@ def get(self):
...
+        if not isinstance(password, str):
+            abort(HTTP_STATUS_CODE_FORBIDDEN, "Password needs to be a string.")
+
+        user = User.get_or_create(username=username, name=username)
+        user.set_password(plaintext=password)
+        # TODO: Take additional attributes of users into account
+        db_session.add(user)
+        db_session.commit()
+
+        user = User.get_or_create(username=username)
```
Comment (berggren) [inline timesketch/api/v1/resources/user.py:90]:
```suggestion
```
You don't have to fetch the User object again, you do it above.

File: timesketch/api/v1/resources/user.py [hunk 5]
```diff
@@ -46,6 +47,84 @@ def get(self):
...
+        return_user = {
+            "id": user.id,
+            "username": user.username,
+            "name": user.name,
+            "email": user.email,
+            "active": user.active,
+            "admin": user.admin,
+        }
+
+        return jsonify({"objects": [return_user]})
```
Comment (berggren) [inline timesketch/api/v1/resources/user.py:101]:
```suggestion
return self.to_json(user)
```
You have added the fields to the resource field mapping in __init__. You just need to call self.to_json here

File: timesketch/api/v1/resources/user.py [hunk 6]
```diff
@@ -46,6 +47,84 @@ def get(self):
...
```
Comment (berggren) [inline timesketch/api/v1/resources/user.py:126]:
```suggestion
return self.to_json(user)
```

**Review by berggren — CHANGES_REQUESTED** [review]
PTAL

**Review by berggren — APPROVED** [review]
LGTM

**Comment by berggren** [comment]
Thanks for contributing!

**Comment by lo-chr** [comment]
Thank your for the reviews and advice. I learned one or two things. ☺️

**Pull request merged**